explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
+* To provide an arbitrary image badge. The web platform already provides this capability via favicons.
 
-* Nothing. Apply the badge for a less specific scope, or, if there is no matching badge, display nothing.
-* A "flag" indicating the presence of a badge with no contents, or
-* A positive integer.
+Possible areas for expansion:
 
-The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).
+* Support rendering a small status indicator (e.g., a music app shows ▶️ or ⏸️; a weather app shows ⛈️ or ⛅️): either a pre-defined set of glyphs or simply allowing a Unicode character to be rendered.
 
-### The API
+Examples of sites that may use this API:
 
-The `Badge` interface is a member object on
-[`Window`](https://html.spec.whatwg.org/#the-window-object). It contains the following methods:
+* Chat, email, and social apps could signal that new messages have arrived.
+* Any application that needs to signal that user action is required (e.g., in a turn-based game, when it is the player's turn).
+* As a permanent indicator of a page's status (e.g., on a build page, to show that the build has completed).
 
-```webidl
-dictionary BadgeOptions {
-  // The scope the badge applies to.
-  // If unspecified this is the origin of the current page.
-  USVString scope;
-}
+### Use with Notifications
 
-[Exposed=Window]
-interface Badge {
-  // Sets the badge for a scope to be |contents|.
-  void set(unsigned long long contents, optional BadgeOptions options);
+The Badge API goes hand-in-hand with Notifications, since both APIs provide a
+way to give the user updates when they aren't directly looking at the page.
+Arguably, Badge could be part of Notifications, but we view it as a separate,
+complementary API.
 
-  // Sets the badge for a scope to be the special value, 'flag'.
-  void set(optional BadgeOptions options);
+Advantages of using the badging API over notifications:
 
-  // Clears the badge for a scope.
-  void clear(optional BadgeOptions options);
-}
-```
+* Can be used for much higher frequency / lower priority events than notifications, because each new event does not disrupt the user.
+* There may be no need to request permission to use the badging API, since it is much less invasive than a notification.
 
-> Note: Should we have a separate overload for boolean flags now, as discussed in [Issue 19](https://github.com/WICG/badging/issues/19) and [Issue 42](https://github.com/WICG/badging/issues/42)?
+Typically, sites will want to use both APIs together: notifications for high-importance events such as new direct messages or incoming calls, and badges for all new messages including group chats not directly addressed to the user.
 
-> Note: This API can only be used from a foreground page. Use from a service worker is being considered for the future. [The FAQ](#Why-cant-this-be-used-in-the-background-from-the-ServiceWorker-see-28-and-5) has more details).
```

Comment:
Do we want some kind of comment to this end still? 

In the short term, we aren't going to support badging from a push event, but you'll still be able to use the API from the ServiceWorker (background sync, fetch, whatever).

Reply:
Yes, there's a lot about this in the Push section. I think it used to be misleading that we said not to use from a SW, when we really meant that you couldn't use it from a push without also showing a notification. Let's not tie those two restrictions together.